Automattic/wp-calypso
# Purchases: Make Jetpack Search Free removable

#### Proposed Changes

- Codifies the new free tier of Jetpack Search into the Calypso codebase ("Jetpack Search Free").
- Enable removing Jetpack Search Free from the purchase management page.

See:
- p1664891351913239-slack-C0117V2PCAE
- pcNPJE-145-p2#comment-1770.

#### Testing Instructions


Testing this feature requires purchasing a Jetpack Search Free subscription for a site. Here's how you can spin up a test Jetpack site and purchase a subscription:

* Use [this link to spin up a Jurassic Ninja **without the Jetpack plugin** but **with the Jetpack Beta plugin**](https://href.li/?https://jurassic.ninja/create/?nojetpack&jetpack-beta).
* Navigate to the Jetpack Beta tab on your JN site and enable the Jetpack Search plugin at "bleeding edge".
* Navigate to the Jetpack Search dashboard on your JN site: `/wp-admin/admin.php?page=jetpack-search&new_pricing_202208=1`. Note the special query string value used to enable the new free tier experience.
* Click on "Start for free" and complete the connection & checkout process. Once this is complete, you should be redirected to the Search dashboard.

Testing changes from this PR:
* Navigate to your test site's `/purchases/subscriptions/` on Calypso.
* Select the Jetpack Search Free subscription.
* Ensure that a "Remove subscription" button appears at the bottom of the page.
* Ensure that the said button works as expected.

<img width="1064" alt="image" src="https://user-images.githubusercontent.com/4044428/194969294-2c3cffca-6eab-4bfd-a0c0-73b4efe1ae8a.png">


#### Pre-merge Checklist
